cvat/apps/dataset_manager/bindings.py

```diff
@@ -457,17 +483,42 @@ def match_frame_fuzzy(self, path):
         return None
 
 class CvatTaskDataExtractor(datumaro.SourceExtractor):
-    def __init__(self, task_data, include_images=False):
+    def __init__(self, task_data, include_images=False, format_type=None, dimension=DimensionType.DIM_2D):
         super().__init__()
-        self._categories = self._load_categories(task_data)
-
+        self._categories, self._user = self._load_categories(task_data, dimension=dimension)
+        self._dimension = dimension
+        self._format_type = format_type
         dm_items = []
 
+        if format_type == "point_cloud":
+            dm_item = self.dm_item(task_data)
+
         is_video = task_data.meta['task']['mode'] == 'interpolation'
         ext = ''
         if is_video:
             ext = FrameProvider.VIDEO_FRAME_EXT
-        if include_images:
+
+        if dimension == DimensionType.DIM_3D:
+            def _make_image(image_id, **kwargs):
+                loader = osp.join(
+                    task_data.db_task.data.get_upload_dirname(), kwargs['path'])
+                related_images = []
+                image = Img.objects.get(id=image_id)
```

Comment:
Is this code going to make a db request per each image accessed?

Reply:
Yes, it will